internal/controller/hooks/check_hook.go

```diff
@@ -162,65 +168,124 @@ func getResourcesList(k8sReader client.Reader, hook *kubeobjects.HookSpec, log l
 	if hook.LabelSelector != nil {
 		log.Info("getting resources using labelSelector", "labelSelector", hook.LabelSelector)
 
-		err := getResourcesUsingLabelSelector(k8sReader, hook, objList)
+		err := getResourcesUsingLabelSelector(k8sReader, hook, uList)
 		if err != nil {
 			return resourceList, fmt.Errorf("error getting resources using labelSelector: %w", err)
 		}
 
-		objsUsingLabelSelector := getObjectsBasedOnType(objList)
+		objsUsingLabelSelector := getObjectsBasedOnType(uList)
 		resourceList = append(resourceList, objsUsingLabelSelector...)
 	}
 
 	return resourceList, nil
 }
 
-func getObjectListBasedOnResourceType(selectResource string) (client.ObjectList, error) {
-	switch selectResource {
-	case podType:
-		return &corev1.PodList{}, nil
-	case deploymentType:
-		return &appsv1.DeploymentList{}, nil
-	case statefulsetType:
-		return &appsv1.StatefulSetList{}, nil
-	default:
-		return nil, fmt.Errorf("unsupported resource type %s", selectResource)
+func validateAndGetUnstructedListBasedOnType(resourceType string) (*unstructured.UnstructuredList, error) {
+	resourceParts := strings.Split(resourceType, "/")
+	if len(resourceParts) != 1 && len(resourceParts) != three {
```

Comment:
Maybe you can prepare list of supported resources and just check, if resourceParts are correlating with this list? I honestly don't think current check is clear enough.